tests/phpunit/tests/general/wpTitle.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/**
+ * @group general
+ * @group template
+ * @covers ::wp_title
+ */
+class Tests_General_WpTitle extends WP_UnitTestCase {
+
+	/**
+	 * @ticket 31521
+	 *
+	 * @dataProvider data_wp_title_archive
+	 */
+	public function test_wp_title_archive( $query, $expected ) {
+		self::factory()->post->create(
+			array(
+				'post_status' => 'publish',
+				'post_title'  => 'Test Post',
+				'post_type'   => 'post',
+				'post_date'   => '2021-11-01 18:52:17',
+			)
+		);
+		$this->go_to( '?m=' . $query );
+
+		$this->assertSame( $expected, wp_title( '&raquo;', false ) );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array
+	 */
+	public function data_wp_title_archive() {
+		return array(
+			'year with posts'                => array(
+				'query'    => '2021',
+				'expected' => ' &raquo; 2021',
+			),
```

Comment:
When the patch is not applied, this data set fails (which confirms the bug):

```
E.....                                                              6 / 6 (100%)

Time: 00:00.320, Memory: 178.50 MB

There was 1 error:

1) Tests_General_WpTitle::test_wp_title_archive with data set "year with posts" ('2021', ' &raquo; 2021')
Undefined index: 00

/var/www/src/wp-includes/class-wp-locale.php:287
/var/www/src/wp-includes/general-template.php:1379
/var/www/tests/phpunit/tests/general/wpTitle.php:26
```

It fails because no month is passed to it. Therefore, an empty string is passed to `$wp_locale->get_month()` which requires a `'00'` to `'12'` value.